### PR TITLE
[REVIEW] Pin max `dask` and `distributed` versions to `2021.09.1`

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR pins max `dask` & `distributed` versions to `2021.09.1` for `21.10` release.

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

File: conda/environments/cudf_dev_cuda11.0.yml [hunk 1]
```diff
@@ -39,8 +39,8 @@ dependencies:
   - mypy=0.782
   - typing_extensions
   - pre_commit
-  - dask>=2021.6.0
-  - distributed>=2021.6.0
+  - dask>=2021.6.0,<=2021.09.1
+  - distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/environments/cudf_dev_cuda11.0.yml:43]:
```suggestion
  - dask=2021.9.1
  - distributed=2021.9.1
```

File: conda/environments/cudf_dev_cuda11.2.yml [hunk 2]
```diff
@@ -39,8 +39,8 @@ dependencies:
   - mypy=0.782
   - typing_extensions
   - pre_commit
-  - dask>=2021.6.0
-  - distributed>=2021.6.0
+  - dask>=2021.6.0,<=2021.09.1
+  - distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/environments/cudf_dev_cuda11.2.yml:43]:
```suggestion
  - dask=2021.9.1
  - distributed=2021.9.1
```

File: conda/recipes/custreamz/meta.yaml [hunk 3]
```diff
@@ -31,8 +31,8 @@ requirements:
     - python
     - streamz 
     - cudf {{ version }}
-    - dask>=2021.6.0
-    - distributed>=2021.6.0
+    - dask>=2021.6.0,<=2021.09.1
+    - distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/recipes/custreamz/meta.yaml:35]:
```suggestion
    - dask=2021.9.1
    - distributed=2021.9.1
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 4]
```diff
@@ -26,13 +26,13 @@ requirements:
   host:
     - python
     - cudf {{ version }}
-    - dask>=2021.6.0
-    - distributed>=2021.6.0
+    - dask>=2021.6.0,<=2021.09.1
+    - distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/recipes/dask-cudf/meta.yaml:30]:
```suggestion
    - dask=2021.9.1
    - distributed=2021.9.1
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 5]
```diff
@@ -26,13 +26,13 @@ requirements:
...
   run:
     - python
     - cudf {{ version }}
-    - dask>=2021.6.0
-    - distributed>=2021.6.0
+    - dask>=2021.6.0,<=2021.09.1
+    - distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline conda/recipes/dask-cudf/meta.yaml:35]:
```suggestion
    - dask=2021.9.1
    - distributed=2021.9.1
```

File: python/custreamz/dev_requirements.txt [hunk 6]
```diff
@@ -3,8 +3,8 @@
 flake8==3.8.3
 black==19.10b0
 isort==5.6.4
-dask>=2021.6.0
-distributed>=2021.6.0
+dask>=2021.6.0,<=2021.09.1
+distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline python/custreamz/dev_requirements.txt:7]:
```suggestion
dask==2021.9.1
distributed==2021.9.1
```

File: python/dask_cudf/dev_requirements.txt [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # Copyright (c) 2021, NVIDIA CORPORATION.
 
-dask>=2021.6.0
-distributed>=2021.6.0
+dask>=2021.6.0,<=2021.09.1
+distributed>=2021.6.0,<=2021.09.1
```
Comment (galipremsagar) [inline python/dask_cudf/dev_requirements.txt:4]:
```suggestion
dask==2021.9.1
distributed==2021.9.1
```

File: python/dask_cudf/setup.py [hunk 8]
```diff
@@ -10,8 +10,8 @@
 
 install_requires = [
     "cudf",
-    "dask>=2021.6.0",
-    "distributed>=2021.6.0",
+    "dask>=2021.6.0,<=2021.09.1",
+    "distributed>=2021.6.0,<=2021.09.1",
```
Comment (galipremsagar) [inline python/dask_cudf/setup.py:14]:
```suggestion
    "dask==2021.9.1",
    "distributed==2021.9.1",
```

File: python/dask_cudf/setup.py [hunk 9]
```diff
@@ -23,8 +23,8 @@
         "pandas>=1.0,<1.4.0dev0",
         "pytest",
         "numba>=0.53.1",
-        "dask>=2021.6.0",
-        "distributed>=2021.6.0",
+        "dask>=2021.6.0,<=2021.09.1",
+        "distributed>=2021.6.0,<=2021.09.1",
```
Comment (galipremsagar) [inline python/dask_cudf/setup.py:27]:
```suggestion
        "dask==2021.9.1",
        "distributed==2021.9.1",
```

File: ci/gpu/build.sh [hunk 10]
```diff
@@ -101,8 +101,8 @@ function install_dask {
     # Install the main version of dask, distributed, and streamz
     gpuci_logger "Install the main version of dask, distributed, and streamz"
     set -x
-    pip install "git+https://github.com/dask/distributed.git@2021.07.1" --upgrade --no-deps
-    pip install "git+https://github.com/dask/dask.git@2021.07.1" --upgrade --no-deps
+    pip install "git+https://github.com/dask/distributed.git@2021.09.1" --upgrade --no-deps
+    pip install "git+https://github.com/dask/dask.git@2021.09.1" --upgrade --no-deps
```
Comment (jakirkham) [inline ci/gpu/build.sh:105]: Would it be worth grouping these into an environment variable that we set once?

File: ci/benchmark/build.sh [hunk 11]
```diff
@@ -75,10 +75,10 @@ conda install "rmm=$MINOR_VERSION.*" "cudatoolkit=$CUDA_REL" \
...
 
 # Install the master version of dask, distributed, and streamz
-logger "pip install git+https://github.com/dask/distributed.git@main --upgrade --no-deps"
-pip install "git+https://github.com/dask/distributed.git@main" --upgrade --no-deps
-logger "pip install git+https://github.com/dask/dask.git@main --upgrade --no-deps"
-pip install "git+https://github.com/dask/dask.git@main" --upgrade --no-deps
+logger "pip install git+https://github.com/dask/distributed.git@2021.09.1 --upgrade --no-deps"
+pip install "git+https://github.com/dask/distributed.git@2021.09.1" --upgrade --no-deps
+logger "pip install git+https://github.com/dask/dask.git@2021.09.1 --upgrade --no-deps"
+pip install "git+https://github.com/dask/dask.git@2021.09.1" --upgrade --no-deps
```
Comment (jakirkham) [inline ci/benchmark/build.sh:81]: Also an environment variable might help here

**Review by jakirkham — COMMENTED** [review]
Thanks Prem! 😄 

Had a couple minor suggestions above

File: conda/environments/cudf_dev_cuda11.0.yml [hunk 12]
```diff
@@ -39,8 +39,8 @@ dependencies:
   - mypy=0.782
   - typing_extensions
   - pre_commit
-  - dask>=2021.6.0
-  - distributed>=2021.6.0
+  - dask=2021.9.1
+  - distributed=2021.9.1
```
Comment (galipremsagar) [inline conda/environments/cudf_dev_cuda11.0.yml:43]:
```suggestion
  - dask=2021.09.1
  - distributed=2021.09.1
```

File: conda/environments/cudf_dev_cuda11.2.yml [hunk 13]
```diff
@@ -39,8 +39,8 @@ dependencies:
   - mypy=0.782
   - typing_extensions
   - pre_commit
-  - dask>=2021.6.0
-  - distributed>=2021.6.0
+  - dask=2021.9.1
+  - distributed=2021.9.1
```
Comment (galipremsagar) [inline conda/environments/cudf_dev_cuda11.2.yml:43]:
```suggestion
  - dask=2021.09.1
  - distributed=2021.09.1
```

File: conda/recipes/custreamz/meta.yaml [hunk 14]
```diff
@@ -31,8 +31,8 @@ requirements:
     - python
     - streamz 
     - cudf {{ version }}
-    - dask>=2021.6.0
-    - distributed>=2021.6.0
+    - dask=2021.9.1
+    - distributed=2021.9.1
```
Comment (galipremsagar) [inline conda/recipes/custreamz/meta.yaml:35]:
```suggestion
    - dask=2021.09.1
    - distributed=2021.09.1
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 15]
```diff
@@ -26,13 +26,13 @@ requirements:
   host:
     - python
     - cudf {{ version }}
-    - dask>=2021.6.0
-    - distributed>=2021.6.0
+    - dask=2021.9.1
+    - distributed=2021.9.1
```
Comment (galipremsagar) [inline conda/recipes/dask-cudf/meta.yaml:30]:
```suggestion
    - dask=2021.09.1
    - distributed=2021.09.1
```

File: conda/recipes/dask-cudf/meta.yaml [hunk 16]
```diff
@@ -26,13 +26,13 @@ requirements:
...
   run:
     - python
     - cudf {{ version }}
-    - dask>=2021.6.0
-    - distributed>=2021.6.0
+    - dask=2021.9.1
+    - distributed=2021.9.1
```
Comment (galipremsagar) [inline conda/recipes/dask-cudf/meta.yaml:35]:
```suggestion
    - dask=2021.09.1
    - distributed=2021.09.1
```

File: python/custreamz/dev_requirements.txt [hunk 17]
```diff
@@ -3,8 +3,8 @@
 flake8==3.8.3
 black==19.10b0
 isort==5.6.4
-dask>=2021.6.0
-distributed>=2021.6.0
+dask==2021.9.1
+distributed==2021.9.1
```
Comment (galipremsagar) [inline python/custreamz/dev_requirements.txt:7]:
```suggestion
dask==2021.09.1
distributed==2021.09.1
```

File: python/dask_cudf/dev_requirements.txt [hunk 18]
```diff
@@ -1,7 +1,7 @@
 # Copyright (c) 2021, NVIDIA CORPORATION.
 
-dask>=2021.6.0
-distributed>=2021.6.0
+dask==2021.9.1
+distributed==2021.9.1
```
Comment (galipremsagar) [inline python/dask_cudf/dev_requirements.txt:4]:
```suggestion
dask==2021.09.1
distributed==2021.09.1
```

File: python/dask_cudf/setup.py [hunk 19]
```diff
@@ -10,8 +10,8 @@
 
 install_requires = [
     "cudf",
-    "dask>=2021.6.0",
-    "distributed>=2021.6.0",
+    "dask==2021.9.1",
+    "distributed==2021.9.1",
```
Comment (galipremsagar) [inline python/dask_cudf/setup.py:14]:
```suggestion
    "dask==2021.09.1",
    "distributed==2021.09.1",
```

File: python/dask_cudf/setup.py [hunk 20]
```diff
@@ -23,8 +23,8 @@
         "pandas>=1.0,<1.4.0dev0",
         "pytest",
         "numba>=0.53.1",
-        "dask>=2021.6.0",
-        "distributed>=2021.6.0",
+        "dask==2021.9.1",
+        "distributed==2021.9.1",
```
Comment (galipremsagar) [inline python/dask_cudf/setup.py:27]:
```suggestion
        "dask==2021.09.1",
        "distributed==2021.09.1",
```

File: ci/benchmark/build.sh (hunk 11 above)
Comment (galipremsagar) [reply ci/benchmark/build.sh]: Done

File: ci/gpu/build.sh (hunk 10 above)
Comment (galipremsagar) [reply ci/gpu/build.sh]: Sure, added one

Comment (jakirkham) [reply ci/gpu/build.sh]: Thanks Prem! 😄

**Comment by galipremsagar** [comment]
rerun tests

**Review by brandon-b-miller — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**